src/GHC/Util.hs

```diff
@@ -37,40 +37,30 @@ import GHC.Util.RdrName
 import GHC.Util.Scope
 import GHC.Util.Unify
 
+import qualified Language.Haskell.GhclibParserEx.Parse as GhclibParserEx
+
 import HsSyn
 import Lexer
 import Parser
 import SrcLoc
 import FastString
 import StringBuffer
 import GHC.LanguageExtensions.Type
-import Panic
-import HscTypes
-import HeaderInfo
 import DynFlags
 
 import Data.List.Extra
 import System.FilePath
 import Language.Preprocessor.Unlit
 
-parseGhcLib :: P a -> String -> DynFlags -> ParseResult a
-parseGhcLib p str flags =
-  Lexer.unP p parseState
-  where
-    location = mkRealSrcLoc (mkFastString "<string>") 1 1
-    buffer = stringToStringBuffer str
-    parseState = mkPState flags buffer location
-
 parseExpGhcLib :: String -> DynFlags -> ParseResult (LHsExpr GhcPs)
-parseExpGhcLib = parseGhcLib Parser.parseExpression
+parseExpGhcLib = GhclibParserEx.parseExpr
 
 parseImportGhcLib :: String -> DynFlags -> ParseResult (LImportDecl GhcPs)
-parseImportGhcLib = parseGhcLib Parser.parseImport
+parseImportGhcLib = GhclibParserEx.parseImport
 
-parseFileGhcLib :: FilePath
-                -> String
-                -> DynFlags
-                -> ParseResult (Located (HsModule GhcPs))
+-- TODO(SF): GhclibParserEx.parseFile doesn't take 'unlit' into
+-- consideration yet.
```

Comment:
This is OK because we run unlit ourselves before?

Reply:
The functions in this file are working around the inadequacies in the `ghc-lib-parser-ex` versions. When `ghc-lib-parser-ex` catches up, the todos and indeed these functions themselves can go away.

Reply:
The ghc versions of unlit and cpphs both call external tools and were a pig to integrate into ghc-lib. The ones in this package use cpphs which doesn't require external tools, which makes the more robust, doesn't require a GHC installation etc. I'd rather stick with the cpphs ones, and I suspect that should stay in hlint rather than move to GHC.

Reply:
Actually, we've never run `unlit` in the context of this function before neither in hlint or ghc-lib-parser-ex. The context in hlint for calling it is such that we've never needed to. Do you think this family of functions should in general `unlit`?

Reply:
So is the net result saying, let's stay with the status quo and `ghc-lib-parser-ex` doesn't do `unlit`? That would work for me.